.github/scripts/lastMod.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+DATA_STATUS=$(Rscript helper/check_lastModification.R)
+echo "$DATA_STATUS"
+if [ "$DATA_STATUS" = "[1] NULL" ]; then
+    echo "No dataset detected"
+elif [ "$DATA_STATUS" = "[1] FALSE" ]; then
+    echo "Dataset has not been modified" 
+    R CMD build .
+    R CMD INSTALL primace_0.0.0.9000.tar.gz
+    Rscript -e "library(primace)"
```

Comment:
add `library(shiny)` here

Reply:
I think the main problem that cause gh actions workflow fails is because of unnecessary `validate` function in `preprocessing()` function. Beause we have add `tryCatch` in the function which prevent app to be crushed, so we must remove `validate` function from begining of `preprocessing()` function as it cause the webapp to be crushed

Reply:
yes, i think so, i will check it

Reply:
but other funcions may cause other `unknown function` error as well

Reply:
> but other funcions may cause other `unknown function` error as well

yes but lets check it

Reply:
it worked, but we shoud change `read_data()` function in `data` arguments of `preprocessing()` function to `read_data_s3()`, then proceed to analysis on `$train` data

Reply:
> it worked, but we shoud change `read_data()` function in `data` arguments of `preprocessing()` function to `read_data_s3()`, then proceed to analysis on `$train` data

it's ok

Reply:
> > it worked, but we shoud change `read_data()` function in `data` arguments of `preprocessing()` function to `read_data_s3()`, then proceed to analysis on `$train` data
> 
> it's ok

You forgot to add `return()` value of `one_uear_dataset` variable in `error` part of `tryCatch`. So it is not working yet